client/dashboard/me/billing-purchases/change-payment-method.tsx

```diff
@@ -0,0 +1,99 @@
+import {
+	purchaseQuery,
+	userSettingsQuery,
+	userPaymentMethodsQuery,
+	stripeConfigurationQuery,
+	razorpayConfigurationQuery,
+	queryClient,
+} from '@automattic/api-queries';
+import { RazorpayHookProvider } from '@automattic/calypso-razorpay';
+import { StripeHookProvider, useStripe } from '@automattic/calypso-stripe';
+import { useQuery, useSuspenseQuery } from '@tanstack/react-query';
+import { useNavigate } from '@tanstack/react-router';
+import { __experimentalVStack as VStack } from '@wordpress/components';
+import { __ } from '@wordpress/i18n';
+import { useEffect } from 'react';
+import { changePaymentMethodRoute } from '../../app/router/me';
+import { PageHeader } from '../../components/page-header';
+import PageLayout from '../../components/page-layout';
+import { PaymentMethodSelector } from './payment-method-selector';
+import getPaymentMethodIdFromPayment from './payment-method-selector/get-payment-method-id-from-payment';
+import { useCreateAssignablePaymentMethods } from './payment-method-selector/use-create-assignable-payment-methods';
+import { getPurchaseUrl } from './urls';
+
+import './style.scss';
+
+function ChangePaymentMethod() {
+	const { purchaseId } = changePaymentMethodRoute.useParams();
+	const navigate = useNavigate();
+
+	const { data: purchase } = useSuspenseQuery( purchaseQuery( parseInt( purchaseId ) ) );
+	const { isLoading: isLoadingStoredCards } = useQuery(
+		userPaymentMethodsQuery( { type: 'card' } )
+	);
+	const { isStripeLoading } = useStripe();
+
+	const currentPaymentMethodId = getPaymentMethodIdFromPayment( purchase );
+	const changePaymentMethodTitle = getChangePaymentMethodTitleCopy( currentPaymentMethodId );
+	const paymentMethods = useCreateAssignablePaymentMethods();
+
+	const isDataLoading = isLoadingStoredCards || isStripeLoading;
+
+	useEffect( () => {
+		if ( ! isDataLoading && ! purchase ) {
+			// Redirect if invalid data
+			navigate( { to: '/me/billing/purchases' } );
+		}
+	}, [ isDataLoading, purchase, navigate ] );
+
+	if ( isDataLoading || ! purchase ) {
+		return null;
+	}
+
+	const successCallback = () => {
+		navigate( { to: getPurchaseUrl( purchase ) } );
+	};
+
+	return (
+		<PageLayout size="small" header={ <PageHeader title={ changePaymentMethodTitle } /> }>
+			<VStack spacing={ 6 }>
+				<PaymentMethodSelector
+					purchase={ purchase }
+					paymentMethods={ paymentMethods }
+					successCallback={ successCallback }
+				/>
+			</VStack>
+		</PageLayout>
+	);
+}
+
+function getChangePaymentMethodTitleCopy( currentPaymentMethodId: string ): string {
+	if ( [ 'credits', 'none' ].includes( currentPaymentMethodId ) ) {
+		return __( 'Add payment method' );
+	}
+	return __( 'Change payment method' );
+}
+
+export default function ChangePaymentMethodWrapper() {
+	const { data: userSettings } = useQuery( userSettingsQuery() );
+	const locale = userSettings?.language || 'en';
+
+	const fetchStripeConfiguration = async ( requestArgs?: {
```

Comment:
Any reason why we shouldn't make use of `useCallback` for these config functions? Upstream providers could trigger renders in the future.